### PR TITLE
BUG: Removing code that removes field type in set_title() 

### PR DESCRIPTION
**Pull request by chummels**
## PR Summary

Removing code that removes field type in set_title().  Not sure why this was ever here.  It is not present for any other plot annotations.

This resolves Issue #5087 .

### CODE REVIEW

**Comment by neutrinoceros** [comment]
This looks like a breaking change to me, the existing logic looks very much intentional (albeit flawed). Other plot annotations don't need to display a field name so I'm not surprised that it's the only one doing this.

**Comment by chummels** [comment]
I added a test for this.  Previously PhasePlot didn't have any SPH-based tests.  Now it does, and it fails for dev tip but succeeds for this PR.  I don't fully understand why this doesn't raise errors for grid-based datasets, but it doesn't seem to.

**Comment by chummels** [comment]
pre-commit.ci autofix

**Review by chrishavlin — DISMISSED** [review]
looks good to me, and i agree with @chummels it's not a breaking change. It's only changing the key that's used to store the desired plot title.

**Comment by chrishavlin** [comment]
I'll give @neutrinoceros a chance to respond, but will plan to merge later today

File: yt/visualization/profile_plotter.py [hunk 1]
```diff
@@ -1319,8 +1319,6 @@ def annotate_title(self, title):
 
         """
         for f in self._profile.field_data:
-            if isinstance(f, tuple):
-                f = f[1]
```
[the remark this answers is not in the document]
Comment (chummels) [reply yt/visualization/profile_plotter.py:1323]: Yeah, I don't really understand why this was here.  But I figured it was in such a niche corner of the code that no one had run into it except me.  I don't think people often use the `set_title()` functionality, let alone for profiles, and specifically with particle-based datasets.

**Comment by neutrinoceros** [comment]
please give me the night ! I don't remember the details but I do want to have another look tomorrow

**Comment by chrishavlin** [comment]
> please give me the night ! I don't remember the details but I do want to have another look tomorrow

Sure thing! I can also wait a bit longer than tomorrow if you need more time.

**Comment by neutrinoceros** [comment]
Ok I'm willing to accept the change as is, *but* we need to make sure that the image tests are run. As long as the baseline submodule isn't updated, CI is expected to *fail*. The fact that it's passing here indicates that they are actually not picked up at all.

**Comment by neutrinoceros** [comment]
Got it: it's because the module they were added to is explicitly ignored for pytest
https://github.com/yt-project/yt/blob/712b724423a308bf69913f5f83e9f567048ee9a9/pyproject.toml#L384

we need to move the test to a different file and make sure the tests are failing before we can move to the next step, as documented here
https://yt-project.org/docs/dev/developing/testing.html#creating-and-updating-image-baselines-for-pytest-mpl-tests

**Comment by chummels** [comment]
Oh.  I didn't realize there were tests that we had which were being explicitly ignored.  Why is that?  It seems like not a great solution to just not be testing the profiler tests for bugs.

To what test file would you like me to move these tests?

**Comment by neutrinoceros** [comment]
> Oh. I didn't realize there were tests that we had which were being explicitly ignored. Why is that? It seems like not a great solution to just not be testing the profiler tests for bugs.

That's because we're stuck in the middle of a migration from two test frameworks (nose to pytest): pytest doesn't support `yield`-based tests and nose is not maintained anymore (and broken on all versions of Python we still support). We have many tests that can only run on one runner, but we haven't collectively invested the time needed to get rid of nosetest altogether.

> To what test file would you like me to move these tests?

`yt/visualization/tests/test_image_comp_2D_plots.py` would be a nice home to these

**Comment by chummels** [comment]
OK, so I moved the tests to the other file.  Hopefully this is ready to go given that it's a 2-line fix for a simple bug.  Answer tests aren't even really needed--just a unit test.  I just copy-pasted the test from a non-SPH Profile test that included `set_title()`.

**Comment by neutrinoceros** [comment]
Indeed, image test are now properly failing. Can you please make sure that the results match your expectations, and if so, open a PR to https://github.com/yt-project/yt_pytest_mpl_baseline, [as documented](https://yt-project.org/docs/dev/developing/testing.html#creating-and-updating-image-baselines-for-pytest-mpl-tests) ?

**Comment by neutrinoceros** [comment]
Actually I just noticed that the failure mode isn't quite right: it should be failing with "missing baseline", but it's showing changes in *exisiting tests* instead. This is because you apparently reused test names, so the image files are colliding with previous ones. Can you make sure you're using unique test names please ?

**Comment by neutrinoceros** [comment]
Looking at the images more closely, it is indeed not needed to add that many tests, in particular because none of them seems to actually check the problem you are fixing here. A single, targeted image test should be enough, but we do need *some* test for this bug, IMO.

**Comment by chrishavlin** [comment]
> in particular because none of them seems to actually check the problem you are fixing here. A single, targeted image test should be enough, but we do need some test for this bug, IMO.

there's a new `test_phaseplot_set_title` that would cover it, no? And I think the motivation for the additional tests here is that there was no coverage at for the phaseplot functionality with particle data so I think it's helpful to add a bit of extra coverage beyond the fix for the actual bug here. Or are you saying you'd prefer those tests be added as unit tests just to make sure the cases run rather than image comparison tests?

**Comment by chrishavlin** [comment]
the bug wasn't with `set_title(tuple)`, it was when the phaseplot included an ambiguous field (https://github.com/yt-project/yt/issues/5087):

```
import yt
ds = yt.load_sample('FIRE_M12i_ref11')
p = yt.PhasePlot(ds, ('gas', 'density'), ('gas', 'temperature'), [('gas', 'z')], weight_field=None)
p.annotate_title('test')
p.save()
```

**Comment by chrishavlin** [comment]
sorry, meant to explain more but got distracted -- 

in 
```
p = yt.PhasePlot(ds, ('gas', 'density'), ('gas', 'temperature'), [('gas', 'z')], weight_field=None)
p.annotate_title('test')
```
that `('gas', 'z')` becomes just `z` in the `annotate_title` on main becuase it was removing the field type within `annotate_title`. this PR doesn't touch the actual label being used (`'test'`).

e.g., here's the result of running the new test on main:

```python
from yt.testing import fake_random_sph_ds
import yt

ds = fake_random_sph_ds(1000, [[0., 1.], [0., 1.], [0., 1]])
p = yt.PhasePlot(
            ds, ("gas", "density"), ("gas", "density"), ("gas", "mass")
        )   
p.annotate_title("Test Title")
p.render()
```

```python-traceback
yt : [INFO     ] 2025-03-04 10:18:28,564 Parameters: current_time              = 0.0
yt : [INFO     ] 2025-03-04 10:18:28,564 Parameters: domain_dimensions         = [1 1 1]
yt : [INFO     ] 2025-03-04 10:18:28,564 Parameters: domain_left_edge          = [0. 0. 0.]
yt : [INFO     ] 2025-03-04 10:18:28,565 Parameters: domain_right_edge         = [1. 1. 1.]
yt : [INFO     ] 2025-03-04 10:18:28,565 Parameters: cosmological_simulation   = 0
yt : [INFO     ] 2025-03-04 10:18:28,565 Allocating for 1000 particles
Traceback (most recent call last):
  File "/Users/chavlin/src/temp/tempthing.py", line 8, in <module>
    p.annotate_title("Test Title")
  File "/Users/chavlin/src/yt_/yt_dev/yt/yt/visualization/_commons.py", line 112, in newfunc
    retv = f(self, *args, **kwargs)
  File "/Users/chavlin/src/yt_/yt_dev/yt/yt/visualization/profile_plotter.py", line 1324, in annotate_title
    self.plot_title[self.data_source._determine_fields(f)[0]] = title
  File "/Users/chavlin/src/yt_/yt_dev/yt/yt/data_objects/data_containers.py", line 1471, in _determine_fields
    finfo = self.ds._get_field_info(field)
  File "/Users/chavlin/src/yt_/yt_dev/yt/yt/data_objects/static_output.py", line 1001, in _get_field_info
    raise ValueError(
ValueError: The requested field name 'mass' is ambiguous and corresponds to any one of the following field types:
 ['gas', 'io', 'all']
Please specify the requested field as an explicit tuple (<ftype>, <fname>).
```

**Comment by chummels** [comment]
@chrishavlin is correct.  I added the other tests just to ensure particle-based PhasePlots were getting tested for other annotations, just as grid-based PhasePlots are.  I figured I might as well add them while I was updating things here, to potentially avoid other bugs down the line with more coverage.  I didn't think image tests were necessary per se, since unit tests would catch any breaking failures, as they do with this particular bug.  If I want these to just be unit tests, can I just remove the `@pytest.mark.mpl_image_compare` lines from the tests and then just get this merged?  

I'll be honest that I am not really excited about doing all the steps to generate new image test answers for all of this.  It's a two-line bugfix for a simple bug.  So whatever I can do to expedite this and not use any more of anyone's time is what I'd prefer.

**Comment by chrishavlin** [comment]
> If I want these to just be unit tests, can I just remove the @pytest.mark.mpl_image_compare lines from the tests

Yup, that's all you need to do to make them unit tests, but there'd be a couple of cleanup steps I'd recommend: you'd no longer need the test functions to return anything so you could strip out the `return p.plots["gas", "mass"].figure` lines and then I'd also suggest moving the tests to a different file again (sorry!!) since `test_image_comp_2d_plots.py` implies that all the contained tests are image comparisons. and depending on the results of https://github.com/yt-project/yt/pull/5129 I'd actually recommend moving them back to where you had them in the first place (again, sorry!!).

**Comment by chummels** [comment]
OK, I finally got back to this.  I moved the tests back and changed them back to unit tests.  Let me know if there are any other things to do or if this can be merged.

**Comment by chrishavlin** [comment]
looks like you missed a numpy import in the test file when moving things back

**Comment by chummels** [comment]
Oops!  My bad.  I've now added it.  Hopefully the tests complete OK and we can finally call this one done. :)

File: yt/visualization/tests/test_profile_plots.py [hunk 2]
```diff
@@ -66,6 +67,55 @@ def test_phaseplot_set_ylim(self):
...
+        bbox = np.array([[-1.0, 3.0], [1.0, 5.2], [-1.0, 3.0]])
+        cls.ds = fake_random_sph_ds(50, bbox)
+
+    def get_plot(self):
+        return PhasePlot(
+            self.ds, ("gas", "density"), ("gas", "density"), ("gas", "mass")
+        )
+
+    @pytest.mark.parametrize("kwargs", [{}, {"color": "b"}])
+    @pytest.mark.mpl_image_compare
```
Comment (chrishavlin) [inline yt/visualization/tests/test_profile_plots.py:82]:
```suggestion
```

File: yt/visualization/tests/test_profile_plots.py [hunk 3]
```diff
@@ -66,6 +67,55 @@ def test_phaseplot_set_ylim(self):
...
+        p.set_unit(("gas", "mass"), "Msun")
+        p.render()
+
+    @pytest.mark.mpl_image_compare
+    def test_phaseplot_set_xlim(self):
+        p = self.get_plot()
+        p.set_xlim(1e-3, 1e0)
+        p.render()
+
+    @pytest.mark.mpl_image_compare
```
Comment (chrishavlin) [inline yt/visualization/tests/test_profile_plots.py:112]:
```suggestion
```

File: yt/visualization/tests/test_profile_plots.py [hunk 4]
```diff
@@ -66,6 +67,55 @@ def test_phaseplot_set_ylim(self):
...
+        p.set_log(("gas", "mass"), False)
+        p.render()
+
+    @pytest.mark.mpl_image_compare
+    def test_phaseplot_set_unit(self):
+        p = self.get_plot()
+        p.set_unit(("gas", "mass"), "Msun")
+        p.render()
+
+    @pytest.mark.mpl_image_compare
```
Comment (chrishavlin) [inline yt/visualization/tests/test_profile_plots.py:106]:
```suggestion
```

File: yt/visualization/tests/test_profile_plots.py [hunk 5]
```diff
@@ -66,6 +67,55 @@ def test_phaseplot_set_ylim(self):
...
+        p.annotate_title("Test Title")
+        p.render()
+
+    @pytest.mark.mpl_image_compare
+    def test_phaseplot_set_log(self):
+        p = self.get_plot()
+        p.set_log(("gas", "mass"), False)
+        p.render()
+
+    @pytest.mark.mpl_image_compare
```
Comment (chrishavlin) [inline yt/visualization/tests/test_profile_plots.py:100]:
```suggestion
```

File: yt/visualization/tests/test_profile_plots.py [hunk 6]
```diff
@@ -66,6 +67,55 @@ def test_phaseplot_set_ylim(self):
...
+        p.annotate_text(1e-4, 1e-2, "Test text annotation", **kwargs)
+        p.render()
+
+    @pytest.mark.mpl_image_compare
+    def test_phaseplot_set_title(self):
+        p = self.get_plot()
+        p.annotate_title("Test Title")
+        p.render()
+
+    @pytest.mark.mpl_image_compare
```
Comment (chrishavlin) [inline yt/visualization/tests/test_profile_plots.py:94]:
```suggestion
```

File: yt/visualization/tests/test_profile_plots.py [hunk 7]
```diff
@@ -66,6 +67,55 @@ def test_phaseplot_set_ylim(self):
...
+        )
+
+    @pytest.mark.parametrize("kwargs", [{}, {"color": "b"}])
+    @pytest.mark.mpl_image_compare
+    def test_phaseplot_annotate_text(self, kwargs):
+        p = self.get_plot()
+        p.annotate_text(1e-4, 1e-2, "Test text annotation", **kwargs)
+        p.render()
+
+    @pytest.mark.mpl_image_compare
```
Comment (chrishavlin) [inline yt/visualization/tests/test_profile_plots.py:88]:
```suggestion
```

**Review by chrishavlin — COMMENTED** [review]
Also have to remove all the `@pytest.mark.mpl_image_compare` decorator usages, then I think this should be good to go!

**Comment by chrishavlin** [comment]
@chummels I took the liberty of removing the `@pytest.mark.mpl_image_compare` usages that I noted in the previous comment to get this across the line :) I think tests will pass now, but just a heads up you'll want to pull changes locally if you need to make any more edits.

**Comment by chrishavlin** [comment]
pre-commit.ci autofix

**Review by chrishavlin — APPROVED** [review]

**Comment by chrishavlin** [comment]
ok, I think we're good! Going to hit merge since the jenkins test is hanging indefinitely these days (a separate can o' worms).

**Comment by chummels** [comment]
Thanks for getting it across the finish line, Chris!
